prometheus/metrics.go

```diff
@@ -196,12 +179,7 @@ func fetchRange(api prom_v1.API, query string, bounds prom_v1.Range) *Metric {
 	case model.ValMatrix:
 		return &Metric{Matrix: result.(model.Matrix)}
 	}
-	return &Metric{err: fmt.Errorf("Invalid query, matrix expected: %s", query)}
-}
-
-func replaceInvalidCharacters(metricName string) string {
```

Comment:
OK to remove, I guess, although I wonder if this was supposed to have been called from somewhere.

Reply:
It's only usage was removed here: https://github.com/kiali/kiali/pull/776
Probably, it was missed in the cleanup.

Now, I have doubts about why those specific prometheus queries required replacing characters and all other queries don't. Perhaps we should be escaping all other p8s queries? Not sure, pinging @jotak to learn more.

Reply:
yeah this is an old relic of pre-istio-1 era. I can't remember exactly, but yeah, it's safe to remove.